battleship.py

```diff
@@ -89,18 +126,39 @@ def checkShip(grid, ship):
 Returns: 2D list of ints
 '''
 def addShips(grid, numShips):
-    return
-
+    count=0
+    while count<numShips:
+        ship=createShip()
+        if checkShip(grid,ship)==True:
+            for i in range(len(ship)):
+                grid[ship[i][0]][ship[i][1]]=SHIP_UNCLICKED
+            count +=1
+    return grid
 
 '''
 drawGrid(data, canvas, grid, showShips)
 Parameters: dict mapping strs to values ; Tkinter canvas ; 2D list of ints ; bool
 Returns: None
 '''
 def drawGrid(data, canvas, grid, showShips):
+    for i in range(0,data["rows"],1):
+        for j in range(0,data["cols"],1):
+                x1=data["cellsize"]*j
+                y1=data["cellsize"]*i
+                x2=x1+data["cellsize"]
+                y2=y1+data["cellsize"]
+                canvas.create_rectangle(x1,y1,x2,y2,outline='black',fill='blue')
+                if(grid[i][j]==SHIP_UNCLICKED):
+                    x3=data["cellsize"]*j
+                    y3=data["cellsize"]*i
+                    x4=x3+data["cellsize"]
+                    y4=y3+data["cellsize"]
+                    canvas.create_rectangle(x3,y3,x4,y4,outline='black',fill='yellow')

```

Comment:
No need to do the same calculation twice. You can use x1, y1, x2, y2 here also.

Reply:
Modified DrawGrid Function